src/button-group.tsx

```diff
@@ -21,6 +24,15 @@ const ButtonGroup: React.FC<ButtonGroupProps> = ({
     align = 'left',
     dataAttributes,
 }) => {
+    const linkContainerRef = React.useRef<HTMLDivElement>(null);
+    const [hasSmallLink, setHasSmallLink] = React.useState(false);
+
+    React.useLayoutEffect(() => {
+        if (linkContainerRef.current?.querySelector('[data-small-link=true]')) {
```

Comment:
After discussing with @pladaria, this is the only reasonable solution we could come up with. We loose the RSC for ButtonGroup, but we don't think this is serious. The code is very small and it shouldn't have a huge impact

Reply:
Discussed offline. I've updated the logic to avoid these issues. Now it should be ok